pkg/distributor/distributor.go

```diff
@@ -338,97 +370,129 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	}
 	// Count the total samples in, prior to validation or deduplication, for comparison with other metrics.
 	incomingSamples.WithLabelValues(userID).Add(float64(numSamples))
+	// Count the total number of metadata in.
+	incomingMetadata.WithLabelValues(userID).Add(float64(len(req.Metadata)))
 
-	if d.limits.AcceptHASamples(userID) && len(req.Timeseries) > 0 {
-		cluster, replica := findHALabels(d.limits.HAReplicaLabel(userID), d.limits.HAClusterLabel(userID), req.Timeseries[0].Labels)
-		removeReplica, err = d.checkSample(ctx, userID, cluster, replica)
-		if err != nil {
-			if resp, ok := httpgrpc.HTTPResponseFromError(err); ok && resp.GetCode() == 202 {
-				// These samples have been deduped.
-				dedupedSamples.WithLabelValues(userID, cluster).Add(float64(numSamples))
-			}
+	// A WriteRequest can now include samples and metadata. However, at the moment of writing,
```

Comment:
I tend to rely on `git blame` for now but I can your thinking. I proposed changing it:

> A WriteRequest can only contain series or metadata but not both. This might change in the future.

Just enough information to remind us that there might a point in time where this no longer holds true. Which is true for almost everything in the remote write protocol given we don't control it.

Reply:
Changed, unresolve if you'd like me to change/add something.

Reply:
Also, we're having a conversation about this in [this comment](https://github.com/cortexproject/cortex/pull/2336#discussion_r402892237). If you have additional feedback would be great to collate it there.